src/safir/testing/kubernetes.py

```diff
@@ -1010,8 +1518,17 @@ async def list_namespaced_pod(
                     return V1PodList(kind="Pod", items=[])
             else:
                 pods = []
-                for obj in self._objects[namespace]["Pod"].values():
-                    pods.append(obj)
+                if "Pod" in self._objects[namespace]:
+                    for obj in self._objects[namespace]["Pod"].values():
+                        if label_dict:
+                            pod_labels = obj.metadata.labels
+                            for lbl in label_dict:
+                                if (
+                                    lbl not in pod_labels
+                                    or label_dict[lbl] != pod_labels[lbl]
+                                ):
+                                    continue
```

Comment:
Same problem here that this code doesn't do anything. It's probably worth writing a utility function that checks a labels dict against a dict of label requirements, since this same pattern is in four places. (Maybe a utility function that handles both field selectors and label selectors, actually, and incorporates the parsing.)